src/common/snippets/src/lowered/pass/fuse_loops.cpp

```diff
@@ -220,10 +220,11 @@ bool FuseLoops::run(LinearIR& linear_ir, lowered::LinearIR::constExprIt begin, l
                 // Loop_0 (Upper)                 |
                 //   |               =>           |
                 // Loop_1 (Current)     Loop_0 + Loop_1 => new `Loop_1`
-                const auto& input_ports = current_loop_info->get_input_ports();
+                // Make a copy of `input_ports` (not ref), since current_loop_info might be changed and ref will be invalid
+                const auto input_ports = current_loop_info->get_input_ports();
```

Comment:
Could you please explain how `current_loop_info` can be changed? By another thread you mean?

Reply:
I meant this lines
https://github.com/openvinotoolkit/openvino/blob/a7d7ad443e371ef47c2464795f946043ae0bab00/src/common/snippets/src/lowered/pass/fuse_loops.cpp#L254-L260

But after this change you compare `input_ports` with new input ports from updated `current_loop_info`
https://github.com/openvinotoolkit/openvino/blob/a7d7ad443e371ef47c2464795f946043ae0bab00/src/common/snippets/src/lowered/pass/fuse_loops.cpp#L263-L264
Here `input_ports` will be invalid if we save them as reference before

Reply:
Ok, thanks for the explanation 🙂